src/cli/stats-docs.ts

```diff
@@ -0,0 +1,112 @@
+  color: 'red' | 'yellow' | 'green' | 'white' | 'magenta' | 'cyan';
+  items?: any;
+}
+type StatsName = 'operations' | 'refs' | 'tags' | 'externalDocs' | 'pathItems' | 'links' | 'schemas' | 'parameters';
+type IStatsCount = Record<StatsName, IStatsRow>;
+
+const statsCount: IStatsCount = {
+  refs: { metric: '🚗 References', total: 0, color: 'red', items: new Set() },
+  externalDocs: { metric: '📦 External Documents', total: 0, color: 'magenta' },
+  schemas: { metric: '📈 Schemas', total: 0, color: 'white'},
+  parameters: { metric: '👉 Parameters', total: 0, color: 'yellow', items: new Set() },
+  links: { metric: '🔗 Links', total: 0, color: 'cyan', items: new Set() },
+  pathItems: { metric: '➡️ Path Items', total: 0, color: 'green' },
+  operations: { metric: '👷 Operations', total: 0, color: 'yellow' },
+  tags: { metric: '🔖 Tags', total: 0, color: 'white' },
+}
+
+function printStatsTable(statsCount: IStatsCount, entrypoint: string) {
+  process.stderr.write(`Document: ${colors.magenta(entrypoint)} stats:\n\n`);
+  for (const node in statsCount) {
+    const { metric, total, color } = statsCount[node as StatsName];
+    process.stderr.write(colors[color](`${metric}: ${total} \n`));
+  }
+}
+
+export async function handleStats (argv: {
+  config?: string;
+  entrypoint?: string;
+}) {
+  const config: LintConfig | Config = await loadConfig(argv.config);
+  const entrypoint = getFallbackEntryPointsOrExit(argv.entrypoint ? [argv.entrypoint] : [], config)[0];
+  const externalRefResolver = new BaseResolver(config.resolve);
+  const document = (await externalRefResolver.resolveDocument(null, entrypoint)) as Document;
+  const lintConfig: LintConfig = config.lint;
+  const oasVersion = detectOpenAPI(document.parsed);
+  const oasMajorVersion = openAPIMajor(oasVersion);
+  const types = normalizeTypes(
+    lintConfig.extendTypes(
+      oasMajorVersion === OasMajorVersion.Version3 ? Oas3Types : Oas2Types,
+      oasVersion,
+    ),
+  );
+
+  const ctx: WalkContext = {
+    problems: [],
+    oasVersion: oasVersion,
+  }
+
+  const resolvedRefMap = await resolveDocument({
+    rootDocument: document,
+    rootType: types.DefinitionRoot,
+    externalRefResolver,
+  });
+
+  const statVisitor: Oas3Visitor | Oas2Visitor = {
+    ExternalDocs: { leave() {statsCount.externalDocs.total++; }},
+    ref: { enter(ref: OasRef) {statsCount.refs.items.add(ref['$ref']); }},
+    Tag: { enter() { statsCount.tags.total++; }},
+    Link: { leave(link: any) { statsCount.links.items.add(link.operationId); }},
+    PathMap: {
+      leave() {
+        statsCount.parameters.total = statsCount.parameters.items.size;
+        statsCount.refs.total = statsCount.refs.items.size;
+        statsCount.links.total = statsCount.links.items.size;
+      },
+      PathItem: {
+        leave() { statsCount.pathItems.total++; },
+        Operation: { leave() { statsCount.operations.total++; }},
+        Parameter: { leave(parameter: Oas2Parameter | Oas3Parameter) {
```

Comment:
Parameter can be defined on the PathItem and on the Operation. I’m not clear if this catches both of those?

Reply:
It catches both.